### PR TITLE
Use annotations for the http port and path

### PR DESCRIPTION
**Pull request by MichelHollands**
Previously these used labels which cannot contain the `/` character.

### CODE REVIEW

File: pkg/admission/prep_downscale.go [hunk 1]
```diff
@@ -234,3 +251,12 @@ func deny(msg string, args ...any) *v1.AdmissionResponse {
 		},
 	}
 }
+
+func getResourceAnnotations(ctx context.Context, ar v1.AdmissionReview, api kubernetes.Interface) (map[string]string, error) {
+	switch ar.Request.Resource.Resource {
+	case "statefulsets":
+		obj, err := api.AppsV1().StatefulSets(ar.Request.Namespace).Get(ctx, ar.Request.Name, metav1.GetOptions{})
+		return obj.Annotations, err
```
Comment (aknuds1) [inline pkg/admission/prep_downscale.go:259]: I believe `obj` will be nil if an error is returned:
```suggestion
		if err != nil {
				return nil, err
		}
		return obj.Annotations, nil
```

**Review by aknuds1 — CHANGES_REQUESTED** [review]
I think there's a risk of dereferencing a nil pointer.

**Review by aknuds1 — APPROVED** [review]
LGTM

**Review by aknuds1 — CHANGES_REQUESTED** [review]
Could you also add this PR to the following changelog entry, please?

```
* [ENHANCEMENT] Add an `/admission/prepare-downscale` mutating admission webhook that prepares pods for shutdown before downscaling. #47
```

**Review by aknuds1 — APPROVED** [review]
LGTM, thanks for applying my suggestions :)

**Review by colega — APPROVED** [review]

**Comment by aknuds1** [comment]
I will merge after CI finishes.

**Pull request merged**